Fabric.Identity.API/scripts/Install-IdentityProviderSearchService.ps1

```diff
@@ -0,0 +1,85 @@
+        throw
+    }
+    try {
+         $commonConfigStore = Get-DosConfigValues -ConfigStore $configStore -Scope "common"
+         Confirm-SettingIsNotNull -settingName "common.webServerDomain" -settingValue $commonConfigStore.webServerDomain
+         Confirm-SettingIsNotNull -settingName "common.clientEnvironment" -settingValue $commonConfigStore.clientEnvironment
+    }
+    catch {
+        Write-DosMessage -Level "Warning" -Message "It is required to have 'webServerDomain' and 'clientEnvironment' populated in install.config"
+        Start-Sleep 60
+        $retryCount++
+    }
+} while ([string]::IsNullOrEmpty($commonConfigStore.webServerDomain) -or [string]::IsNullOrEmpty($commonConfigStore.clientEnvironment))
+
+# Prompt for the credential if it is null
+if($null -eq $credential)
+{
+  $idpssIisUser = Get-IISAppPoolUser -credential $credential -appName $idpssConfigStore.appName -storedIisUser $idpssConfigStore.iisUser -installConfigPath $configStore.Path -scope $idpssSettingsScope
+  $credential = $idpssIisUser.Credential
+}
+
+$idpssServiceUrl = Get-ApplicationEndpoint -appName $idpssConfigStore.appName -applicationEndpoint $idpssConfigStore.applicationEndPoint -installConfigPath $installConfigPath -scope $idpssSettingsScope -quiet $quiet
+    
+$idpssStandalonePath = ".\Fabric.IdentityProviderSearchService.zip"
+$idpssInstallerPath = "..\WebDeployPackages\Fabric.IdentityProviderSearchService.zip"
+$idpssInstallPackagePath = Get-WebDeployPackagePath -standalonePath $idpssStandalonePath -installerPath $idpssInstallerPath
+
+$secretNoEnc = $commonConfigStore.fabricInstallerSecret -replace "!!enc!!:"
+
+$decryptedSecret = Unprotect-DosInstallerSecret -CertificateThumprint $commonConfigStore.encryptionCertificateThumbprint -EncryptedInstallerSecretValue $secretNoEnc
+
+$registrationApiSecret = Add-IdpssApiResourceRegistration -identityServiceUrl $commonConfigStore.identityService -fabricInstallerSecret $decryptedSecret
+
+$idpssWebDeployParameters = Get-IdpssWebDeployParameters -serviceConfig $idpssConfigStore -commonConfig $commonConfigStore -discoveryServiceUrl $discoveryServiceUrl -noDiscoveryService $noDiscoveryService -credential $credential -registrationApiSecret $registrationApiSecret
+
+$idpssInstallApplication = Publish-DosWebApplication -WebAppPackagePath $idpssInstallPackagePath `
+                      -WebDeployParameters $idpssWebDeployParameters `
+                      -AppPoolName $idpssConfigStore.appPoolName `
+                      -AppPoolCredential $credential `
+                      -AuthenticationType "Anonymous" `
+                      -WebDeploy `
+
+$idpssAppPoolUser = $credential.UserName
+
+$idpssName = "IdentityProviderSearchService"
+
+Add-PermissionToPrivateKey $idpssAppPoolUser $certificates.SigningCertificate read
+
+$idpssDirectory = [io.path]::combine([System.Environment]::ExpandEnvironmentVariables($selectedSite.physicalPath), $idpssName)
+New-AppRoot $idpssDirectory $idpssAppPoolUser
```

Comment:
Can we rename this function to New-LogsDirectoryForApp or something similar? We aren't really creating a new application root here, since that was already done in the Publish step above.

Reply:
Sure.